client/src/app/hooks/table-controls/column/useColumnState.ts

```diff
@@ -0,0 +1,25 @@
+import { useState } from "react";
+
+export interface ColumnState<TColumnKey extends string> {
+  id: TColumnKey;
+  label: string;
+  isVisible: boolean;
+}
+
+export interface IColumnState<TColumnKey extends string> {
+  columns: ColumnState<TColumnKey>[];
+  setColumns: (newColumns: ColumnState<TColumnKey>[]) => void;
+}
+
+interface IColumnStateArgs<TColumnKey extends string> {
+  initialColumns: ColumnState<TColumnKey>[];
+}
+
+export const useColumnState = <TColumnKey extends string>(
+  args: IColumnStateArgs<TColumnKey>
+): IColumnState<TColumnKey> => {
+  const [columns, setColumns] = useState<ColumnState<TColumnKey>[]>(
+    args.initialColumns
+  );
+  return { columns, setColumns };
```

Comment:
And this is where you'd have to add column visibility persistence past the component state level that is used now.  I predict there will be a QE ticket filed in less than 10 minutes of app testing say that the column visibility doesn't persist page page refreshes.

A chat before testing would help.  Maybe even loop docs in to level set expectations as to how long the column management state lives.

Reply:
Added local storage support for the column names.